### PR TITLE
docs(contributing): add more info to the guide

### PR DESCRIPTION
**Pull request by ooliver1**
## Summary

Improve `CONTRIBUTING.md` by moving it to the root and describing:

- documentation
- commit style
- type annotations

## Checklist

- [ ] If code changes were made then they have been tested.
  - [ ] I have updated the documentation to reflect the changes.
  - [ ] I have run `task pyright` and fixed the relevant issues.
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [x] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,276 @@
+# Contributing to nextcord
+
+First off, thanks for taking the time to contribute. It makes the library substantially better. :+1:
+
+The following is a set of guidelines for contributing to the repository. These are guidelines, not hard rules.
+
+## This is too much to read! I want to ask a question
+
+Generally speaking questions are better suited in our resources below.
```
Comment (EmreTech) [inline CONTRIBUTING.md:9]:
```suggestion
Generally asking questions are better suited in our resources below.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+    - Tell us **what actually happens**. What ends up happening in reality? It's not helpful to say "it fails" or "it doesn't work". Say *how* it failed, do you get an exception? Does it hang? How are the expectations different from reality?
+    - Tell us **information about your environment**. What version of nextcord are you using? How was it installed? What operating system are you running on? These are valuable questions and information that we use.
+
+If the bug report is missing this information then it'll take us longer to fix the issue. We will probably ask for clarification, and barring that if no response was given then the issue will be closed.
+
+## Submitting a Pull Request
+
+Submitting a pull request is fairly simple, just make sure it focuses on a single aspect and doesn't manage to have scope creep and it's probably good to go. It would be incredibly lovely if the style is consistent to that found in the project.
+
+Please use the following to get your contributions up to spec
```
Comment (EmreTech) [inline CONTRIBUTING.md:37]:
```suggestion
Please use the following to get your contributions up to spec:
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+
+We use [Black](https://github.com/psf/black) for code formatting, [Isort](https://github.com/pycqa/isort) for import sorting, [Slotscheck](https://github.com/ariebovenberg/slotscheck) and several other [pre-commit hooks](https://github.com/pre-commit/pre-commit-hooks) for linting.
+
+To run all of these but `slotscheck`, you can simply run `task lint` in the root of the project. To run `slotscheck`, you can run `task slotscheck`.
+
+If you would like these to run automatically, you can use `task precommit` to install pre-commit hooks. This will run all of the above on every commit.
+
+## Type Annotations
+
+Mafic uses [Pyright](https://github.com/microsoft/pyright) for type checking. To use it, run `task pyright` in the root directory of the project, or `python -m task pyright` (`py -m` etc) if that does not work.
```
Comment (EmreTech) [inline CONTRIBUTING.md:63]:
```suggestion
Nextcord uses [Pyright](https://github.com/microsoft/pyright) for type checking. To use it, run `task pyright` in the root directory of the project, or `python -m task pyright` (`py -m` etc) if that does not work.
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+
+If type annotations are new to you, here is a [guide](https://decorator-factory.github.io/typing-tips/) to help you with it.
+
+## Commits
+
+Mafic follows the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) style. This means that your commit messages and PR titles should be formatted in a specific way. This is to help us generate changelogs and release notes, whilst also helping us review your pull requests as we can see what each commit does.
```
Comment (EmreTech) [inline CONTRIBUTING.md:69]:
```suggestion
Nextcord follows the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) style. This means that your commit messages and PR titles should be formatted in a specific way. This is to help us generate changelogs and release notes, whilst also helping us review your pull requests as we can see what each commit does.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,276 @@
...
+
+## Documentation
+
+Nextcord uses [Sphinx](https://www.sphinx-doc.org/en/master/) for documentation compilation. Specifically, we use the numpy format ([Reference](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard), [Examples](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html)).
+
+### Building the Docs
+
+To install the documentation dependencies, run `pip install -r docs/requirements.txt` in the root of the project.
+
+To build the docs, run `make html` (`./make html` on Windows, requires [Make](https://www.gnu.org/software/make/) on linux). If you don't have make, you can use `python -m sphinx -b html docs docs/_build/html` instead) in the `docs` directory.
```
Comment (EmreTech) [inline CONTRIBUTING.md:147]: Shouldn't Linux be capitalized here? It's a proper noun.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,276 @@
...
+    ...
+```
+
+## Typed Payloads
+
+We use [TypedDict](https://docs.python.org/3/library/typing.html#typing.TypedDict) to type payloads. This is a dictionary with a specific set of keys, where the values have a specific type.
+
+These are stored in `nextcord/types`, where the file name relates to the `nextcord/` filename.
+
+These are python representations of Discord payloads found in the [Discord API docs](https://discord.dev). [`typing_extensions.NotRequired`](https://peps.python.org/pep-0655/) is used for keys which are omitted, represented by `name?` in the Discord docs. [`typing.Optional`](https://docs.python.org/3/library/typing.html#typing.Optional) is used for keys which may be `null` (`None`), represented by `?type` in the Discord docs.
```
Comment (EmreTech) [inline CONTRIBUTING.md:255]: Python should be capitalized here too.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,276 @@
...
+
+`nextcord.types.snowflake.Snowflake` is used for snowflakes, which are Discord IDs.
```
Comment (EmreTech) [inline CONTRIBUTING.md:257]: Capitalize Snowflake.

**Review by EmreTech — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 5 above)
Comment (ooliver1) [reply CONTRIBUTING.md:147]:
```suggestion
To build the docs, run `make html` (`./make html` on Windows, requires [Make](https://www.gnu.org/software/make/) on Linux). If you don't have make, you can use `python -m sphinx -b html docs docs/_build/html` instead) in the `docs` directory.
```

File: CONTRIBUTING.md (hunk 6 above)
Comment (ooliver1) [reply CONTRIBUTING.md:255]:
```suggestion
These are Python representations of Discord payloads found in the [Discord API docs](https://discord.dev). [`typing_extensions.NotRequired`](https://peps.python.org/pep-0655/) is used for keys which are omitted, represented by `name?` in the Discord docs. [`typing.Optional`](https://docs.python.org/3/library/typing.html#typing.Optional) is used for keys which may be `null` (`None`), represented by `?type` in the Discord docs.
```

File: CONTRIBUTING.md (hunk 7 above)
Comment (ooliver1) [reply CONTRIBUTING.md:257]:
```suggestion
`nextcord.types.snowflake.Snowflake` is used for Snowflakes, which are Discord IDs.
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,276 @@
...
+- [The discussions page](https://github.com/nextcord/nextcord/discussions)
+
+Please try your best not to ask questions in our issue tracker. Most of them don't belong there unless they provide value to a larger audience.
+
+## Good Bug Reports
+
+Please be aware of the following things when filing bug reports.
+
+1. Don't open duplicate issues. Please search your issue to see if it has been asked already. Duplicate issues will be closed.
+2. When filing a bug about exceptions or tracebacks, please include the *complete* traceback. Without the complete traceback the issue might be **unsolvable** and you will be asked to provide more information.
```
Comment (EmreTech) [inline CONTRIBUTING.md:23]:
```suggestion
2. When filing a bug about exceptions, please include the *complete* traceback. Without the complete traceback the issue might be **unsolvable** and you will be asked to provide more information.
```
Aren't exceptions always presented in tracebacks?

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,276 @@
...
+3. Make sure to provide enough information to make the issue workable. The issue template will generally walk you through the process but they are enumerated here as well:
+    - A **summary** of your bug report. This is generally a quick sentence or two to describe the issue in human terms.
+    - Guidance on **how to reproduce the issue**. Ideally, this should have a small code sample that allows us to run and see the issue for ourselves to debug. **Please make sure that the token is not displayed**. If you cannot provide a code snippet, then let us know what the steps were, how often it happens, etc.
+    - Tell us **what you expected to happen**. That way we can meet that expectation.
```
Comment (EmreTech) [inline CONTRIBUTING.md:27]:
```suggestion
    - Tell us **what you expected to happen**, that way we can meet that expectation.
```

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,276 @@
...
+
+- **build**: Changes that affect the build system or external dependencies
+- **ci**: Changes to our CI configuration files and scripts such as GitHub Actions
+- **docs**: Documentation only changes
+- **feat**: A new feature
+- **fix**: A bug fix
+- **perf**: A code change that improves performance
+- **refactor**: A code change that neither fixes a bug nor adds a feature
+- **style**: Changes that do not affect the meaning of the code (white-space, formatting, etc)
+- **test**: Adding missing tests or correcting existing tests
```
Comment (EmreTech) [inline CONTRIBUTING.md:116]: What about `chore`?

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,276 @@
...
+
+## Documentation
+
+Nextcord uses [Sphinx](https://www.sphinx-doc.org/en/master/) for documentation compilation. Specifically, we use the numpy format ([Reference](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard), [Examples](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html)).
+
+### Building the Docs
+
+To install the documentation dependencies, run `pip install -r docs/requirements.txt` in the root of the project.
+
+To build the docs, run `make html` (`./make html` on Windows, requires [Make](https://www.gnu.org/software/make/) on Linux). If you don't have make, you can use `python -m sphinx -b html docs docs/_build/html` instead) in the `docs` directory.
```
Comment (EmreTech) [inline CONTRIBUTING.md:147]:
```suggestion
To build the docs, run `make html` (`./make html` on Windows, requires [Make](https://www.gnu.org/software/make/) on Linux. If you don't have make, you can use `python -m sphinx -b html docs docs/_build/html` instead) in the `docs` directory.
```

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -0,0 +1,276 @@
...
+
+        .. versionchanged:: 2.2
+
+            Added the `token` parameter.
+        """
+```
+
+### Operations
+
+We use [`.. container:: operations`](https://docutils.sourceforge.io/docs/ref/rst/directives.html#container) to describe special methods. This should be below the summary, and above the attributes.
```
Comment (EmreTech) [inline CONTRIBUTING.md:186]:
```suggestion
We use [`.. container:: operations`](https://docutils.sourceforge.io/docs/ref/rst/directives.html#container) to describe special methods. This should be below the summary and above the attributes.
```

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -0,0 +1,276 @@
...
+    ...
+```
+
+## Typed Payloads
+
+We use [TypedDict](https://docs.python.org/3/library/typing.html#typing.TypedDict) to type payloads. This is a dictionary with a specific set of keys, where the values have a specific type.
+
+These are stored in `nextcord/types`, where the file name relates to the `nextcord/` filename.
+
+These are Python representations of Discord payloads found in the [Discord API docs](https://discord.dev). [`typing_extensions.NotRequired`](https://peps.python.org/pep-0655/) is used for keys which are omitted, represented by `name?` in the Discord docs. [`typing.Optional`](https://docs.python.org/3/library/typing.html#typing.Optional) is used for keys which may be `null` (`None`), represented by `?type` in the Discord docs.
```
Comment (EmreTech) [inline CONTRIBUTING.md:255]: For `NotRequired`, shouldn't you link to the Python Docs?

**Review by EmreTech — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 11 above)
Comment (ooliver1) [reply CONTRIBUTING.md:147]: Then theres no closing bracket from `` (`./make html` ``

File: CONTRIBUTING.md (hunk 13 above)
Comment (ooliver1) [reply CONTRIBUTING.md:255]: typing_extensions is not in the docs

Comment (EmreTech) [reply CONTRIBUTING.md:255]: True

File: CONTRIBUTING.md (hunk 8 above)
Comment (ooliver1) [reply CONTRIBUTING.md:23]: I never wrote this, it is from the original contributing.md, but sure.

File: CONTRIBUTING.md (hunk 10 above)
Comment (ooliver1) [reply CONTRIBUTING.md:116]:
```suggestion
- **test**: Adding missing tests or correcting existing tests
- **chore**: Other miscellaneous changes which do not affect users
```

**Review by EmreTech — APPROVED** [review]

File: CONTRIBUTING.md [hunk 14]
```diff
@@ -0,0 +1,298 @@
...
+
+Nextcord uses [Sphinx](https://www.sphinx-doc.org/en/master/) for documentation compilation. Specifically, we use the numpy format ([Reference](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard), [Examples](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html)).
+
+### Building the Docs
+
+To install the documentation dependencies, run `pip install -r docs/requirements.txt` in the root of the project.
+
+To build the docs, run `make html` (`./make html` on Windows, requires [Make](https://www.gnu.org/software/make/) on Linux. If you don't have make, you can use `python -m sphinx -b html docs docs/_build/html` instead) in the `docs` directory.
+
+To view the docs, open `docs/_build/html/index.html` in your browser, or run a development server with `python -m http.server --directory docs/_build/html` and open `localhost:8000` in your browser.
```
Comment (DenverCoder1) [inline CONTRIBUTING.md:151]: `task docs` is the simpler way to do this. It builds the docs, runs it on localhost:8069, and watches for live changes

**Review by DenverCoder1 — APPROVED** [review]

**Pull request merged**